spec/higher_level_api/integration/connection_spec.rb

```diff
@@ -336,13 +336,38 @@
           :password => "bunny_password",
           :vhost    => "bunny_testbed",
           :tls                   => true,
+          :verify_peer           => verify_peer,
           :tls_cert              => "spec/tls/client_certificate.pem",
           :tls_key               => "spec/tls/client_key.pem",
           :tls_ca_certificates   => ["./spec/tls/ca_certificate.pem"])
       end
 
-      it "uses TLS port" do
-        expect(subject.port).to eq tls_port
+      context "peer verification is off" do
+        let(:verify_peer) { false }
+
+        it "uses TLS port" do
+          expect(subject.port).to eq tls_port
+        end
+
+        it "sends the SNI details" do
+          # https://github.com/ruby-amqp/bunny/issues/440
+          subject.start
+          expect(subject.transport.socket.hostname).to eq("127.0.0.1")
```

Comment:
This fails when I run tests locally. Perhaps a better strategy would be checking if the hostname is in a set of `{"127.0.0.1", "::1", Socket.gethostname}`. And even that is likely to be fragile. So I'd probably remove this example entirely.